### ISSUE TITLE
Wrapper install instructions should include distributionSha256Sum directly

### ISSUE THREAD

**Issue by leonard84** [gradle/gradle#12412]
@melix gave a nice [talk at devoxx2019](https://www.youtube.com/watch?v=GWGNp3a3hpk) about security where he recommended to use `distributionSha256Sum` in the `gradle-wrapper.properties`. However, it is quite cumbersome to use it manually.

### Current Behavior

The official site https://gradle.org/install/  just lists 
`./gradlew wrapper --gradle-version=6.2.1 --distribution-type=bin`

You have to search for the checksum,  this already presumes that you know about it at all. On the [download page](https://gradle.org/next-steps/?version=6.2.1&format=bin) you have to abort the download of gradle (we want to use the warpper) and then download an extra file containing the sha256Sum, why is it not displayed here directly? If you search for a longer time you might stumble upon https://gradle.org/release-checksums/ which lists all checksum files directly.

### Expected Behavior

The official site https://gradle.org/install/  command  should include `--gradle-distribution-sha256-sum`

`./gradlew wrapper --gradle-version=6.2.1 --distribution-type=bin 
--gradle-distribution-sha256-sum=a68ca7ba57f3404c3f6fc1f70a02d3a7d78652e6b46bbfaff83fc9a17168c279`

This should be an easy fix and will improve the user security without them having to do anything troublesome.

Furthermore, the [download page](https://gradle.org/next-steps/?version=6.2.1&format=bin) should also display the sha256Sum directly while still allowing the download of the checksum.

Also https://docs.gradle.org/current/userguide/gradle_wrapper.html#sec:upgrading_wrapper doesn't include `--gradle-distribution-sha256-sum` by default, it mentioned further down in the document, altough the displayed sum doesn't correspond to either distribution.

### Context

Related #10551 and #5211

**Comment by marcphilipp** [comment]
@JLLeitschuh Is this something for you?

**Comment by JLLeitschuh** [comment]
@big-guy and I had a chat about this a while ago, this may not be possible to do with our currently release pipeline. We may not know the SHA-256 checksum when building this document.

I agree that this would be an awesome value add, but I don't know how easy it would be to actually do it.

**Comment by leonard84** [comment]
Maybe you could do a post processing after the value is known.

**Comment by szpak** [comment]
I've bumped into that issue trying to report a similar one on my own. As an alternative approach (or maybe complementary) the wrapper task could be enhanced with the way to generate sha265sum on upgrade. See #12785.

**Comment by leonard84** [comment]
Another possibility would be to include a small javascript on the page that dynamically retrieves the SHA from the url `https://services.gradle.org/distributions/gradle-6.2.1-bin.zip.sha256`. 
Furthermore, the wrapper could do it during the update process, just retrieve the value and then write it into the `distributionSha256Sum` field of the properties.

**Comment by JLLeitschuh** [comment]
Assuming that we don't generate PDF copies of those documents, that could work.

**Comment by melix** [comment]
>  include a small javascript on the page that dynamically retrieves the SHA from the url 

Security-wise I don't think we should do this. checksums publicized on the documentation should be published independently. If the `services.gradle.org` server is compromised, you would end up displaying the compromised checksum. By separating the two and having separate publication processes, an attacker would have to compromise both the server hosting the artifacts _and_ the server hosting the documentation, which is arguably much more complicated.

**Comment by leonard84** [comment]
@melix I agree with you in principle, but it is better as what we currently have, which is so hidden that practically nobody uses it. Even if you have the `distributionSha256Sum` in you current file, when you update via the officially recommended process `./gradlew wrapper --gradle-version=6.2.1 --distribution-type=bin` you'll loose it.
So my second suggestion was a pragmatic compromise, since the good solution seems to be so much effort that it will not be implemented any time soon. It is not perfect and has the issues you described, but the alternative is that you don't have a `distributionSha256Sum` at all and it would still suffice for the attacker to compromise `services.gradle.org`.

> Assuming that we don't generate PDF copies of those documents, that could work.

Well those PDF would just not include the  `distributionSha256Sum` so it would be no change to the status quo.

**Comment by szpak** [comment]
It seems to be still not implemented.
